### PR TITLE
Add support for `google_compute_autoscaler` to TGC

### PR DESCRIPTION
**Pull request by delimaneto**
adding support for compute.googleapis.com/Autoscaler
```release-note:none
```

### CODE REVIEW

**Comment by slevenick** [comment]
Waiting on discussion with @rileykarson to review this

**Comment by rileykarson** [comment]
These are unblocked, see https://github.com/GoogleCloudPlatform/magic-modules/pull/10540#pullrequestreview-2044173640 for guidance (this appears to follow it)

I kicked the build @slevenick

File: mmv1/third_party/tgc/tests/data/example_google_compute_autoscaler.tf [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  required_providers {
+    google = {
+      source = "hashicorp/google-beta"
+      version = "~> {{.Provider.version}}"
+    }
+  }
+}
+
+provider "google" {
+  {{if .Provider.credentials }}credentials = "{{.Provider.credentials}}"{{end}}
```
Comment (slevenick) [inline mmv1/third_party/tgc/tests/data/example_google_compute_autoscaler.tf:27]: Looks like something about the credentials is failing in our CI test. Are you able to run this locally?

Comment (delimaneto) [reply mmv1/third_party/tgc/tests/data/example_google_compute_autoscaler.tf:27]: Basically I ran gcloud auth application-default login and got the `path/application_default_credentials.json` file and replaced the `GOOGLE_OAUTH_ACCESS_TOKEN=fake-token` to `GOOGLE_APPLICATION_CREDENTIALS=path/application_default_credentials.json` to run locally this case.

Comment (slevenick) [reply mmv1/third_party/tgc/tests/data/example_google_compute_autoscaler.tf:27]: Hm, we're seeing the following error when running these tests:
            │ More details:
            │ Reason: authError, Message: Invalid Credentials

Comment (delimaneto) [reply mmv1/third_party/tgc/tests/data/example_google_compute_autoscaler.tf:27]: I updated the .tf file with a little change. It should work now.

**Review by sushant00 — APPROVED** [review]

**Review by slevenick — APPROVED** [review]

**Pull request merged**